chocolatey/choco
# (#1392) Update refreshenv to detect when not in cmd shell

## Description Of Changes

Update the `refreshenv.cmd` file to detect if being run from a sub shell and properly report that it doesn't work in that scenario.

## Motivation and Context

By default, if you don't have a PowerShell profile when Chocolatey is first installed, you don't get the `refreshenv` alias from the `chocolateyProfile.psm1` PowerShell module. This means if you run `refreshenv` while in PowerShell it actually runs `refreshenv.cmd` and fails to refresh the environment.

## Testing

1. Build the solution with `./build.bat`
2. Install the built files `choco upgrade chocolatey -pre -y -s ./code_drop/Packages/Chocolatey`
3. Run `refreshenv` in a PowerShell session
4. Verify output regarding not supported there
5. Import module referenced in output
6. Run `refreshenv` again
7. Verify it works as expected
8. Run `refreshenv` from a `cmd.exe` session
9. Verify output of refreshing variables for cmd.exe

<img width="793" alt="2022-07-01_09-38-14" src="https://user-images.githubusercontent.com/30301021/176935801-8675f0a7-304f-4863-9579-5c9adf64f1fe.png">


## Change Types Made


## Related Issue

Fixes #1392 

## Change Checklist
